homeassistant/components/gogogate2/config_flow.py

```diff
@@ -0,0 +1,57 @@
+"""Config flow for Gogogate2."""
+import logging
+import re
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PASSWORD, CONF_USERNAME
+
+from .common import async_api_info_or_none, get_api
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class Gogogate2FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Gogogate2 config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_import(self, config_data: dict = None):
+        """Handle importing of configuration."""
+        return await self.async_step_finish(config_data)
+
+    async def async_step_user(self, user_input: dict = None):
+        """Handle user initiated flow."""
+        if user_input is not None:
+            return await self.async_step_finish(user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_IP_ADDRESS): str,
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+        )
+
+    async def async_step_finish(self, user_input: dict):
+        """Validate and create config entry."""
+
+        api = get_api(user_input)
+        data = await async_api_info_or_none(self.hass, api)
+        if data is None:
```

Comment:
The standard is to show them the form again with `errors` filled in so the user can try again if they enter any data such as the ip address incorrectly.  

For an example, see `harmony`

Reply:
I pushed a change to ensure user config flows do not abort. I'm having problems with it not showing the errors ("cannot_connect", "invalid_auth"). I pass the errors to aync_show_form but the errors never appear in the UI. Do you know how to get the errors to show?

Reply:
`cp strings.json translations/en.json`